DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestForControllers/AccountControllerAssignRolesToUserTests.cs

```diff
@@ -0,0 +1,165 @@
+                { 
+                    new List<User>(),
+                    12, 
+                    15, 
+                    "UserRole" 
+                };
+            }
+        }
+
+        public static IEnumerable<object[]> GetDataForInvalidModelStateResult
+        {
+            get
+            {
+                yield return new object[] 
+                { 
+                    new List<User>
+                    {
+                        new User { ProviderId = 1},
+                    },
+                    new List<SecurityRole>
+                    {
+                        new SecurityRole { Name = "Admin"}
+                    },
+                    1,
+                    0, 
+                    "BussinessAnalytics" 
+                };
+            }
+        }
+
+        public static IEnumerable<object[]> GetDataForOkResult
+        {
+            get
+            {
+                yield return new object[]
+                { 
+                    new List<User>
+                    {
+                        new User { ProviderId = 2},
+                    },
+                    new List<SecurityRole>
+                    {
+                        new SecurityRole { Name = "Admin"}
+                    },
+                    new User
+                    {
+                        ProviderId = 1
+                    },
+                    1, 
+                    0, 
+                    "Admin" 
+                };
+            }
+        }
+
+        private ApplicationUserManager MockForUsers(List<User> users)
+        {
+            var userManager = Substitute.For<ApplicationUserManager>(Substitute.For<IUserStore<User>>());
+            //userManager.Users.Returns(new MockForEnumerableQuery<User>(users));
+            userManager.Users.Returns(users.AsQueryable());
+            return userManager;
+        }
+
+        private ISecurityContext MockForContext(IEnumerable<SecurityRole> roles)
+        {
+            var context =Substitute.For<ISecurityContext>();
+            context.SecurityRoles.Returns(new MockForDbSet<SecurityRole>(roles));
+            return context;
+        } 
+
+        private ApplicationUserManager MockForAddingClaim(List<User> users, User userToUpdate)
+        {
+            var userManager = MockForUsers(users);
+            userManager.UpdateAsync(userToUpdate).Returns(IdentityResult.Success);
+            userManager.CreateIdentityAsync(userToUpdate, DefaultAuthenticationTypes.ApplicationCookie).Returns(Task.FromResult(new ClaimsIdentity()));
+            userManager.AddClaimAsync(userToUpdate.Id, Arg.Any<Claim>()).Returns(IdentityResult.Success);
+            return userManager;
+        }
+
+        [Theory]
+        [MemberData("GetDataForNotFountResult")]
+        public void NotFoundUserTest(List<User> users, int gitHubId, int repoId, string roleToAssign)
+        {
+            AccountController controller = new AccountController(Substitute.For<IGithubService>(),
```

Comment:
Please use // Arrange, // Act, // Assert comments to split different phases in unit tests.